web/src/common/Intersection.ts

```diff
@@ -0,0 +1,42 @@
+import type { Feature, Point } from "geojson";
+
+export type IntersectionId = number;
+
+export type DiagonalFilter = {
+  angle: number;
+};
+
+export type IntersectionProperties = {
+  kind: "editable_intersection";
+  intersection_id: number;
+  filter?: DiagonalFilter;
+};
+export type IntersectionFeature = Feature<Point, IntersectionProperties>;
+
+export class Intersection {
+  feature: IntersectionFeature;
+  constructor(feature: IntersectionFeature) {
+    this.feature = feature;
+  }
+
+  get intersectionId(): IntersectionId {
+    return this.feature.properties.intersection_id;
+  }
+
+  get filter(): DiagonalFilter | undefined {
+    return this.feature.properties.filter;
+  }
+
+  get filterAngle(): number | undefined {
+    return this.feature.properties.filter?.angle;
+  }
+
+  get hasAlreadyRotatedFilter(): boolean {
+    if (this.filterAngle === undefined) {
+      return false;
+    }
+    // This is potentially incorrect, if the roads aren't (approximately) evenly distributed.
```

Comment:
Ah, this might be the bug. If I force this to false always, then I can get to the second rotation state (but of course then not delete or keep toggling through the 3 states).

That `split_offset` in the backend was 0 or 1always, right? Could we track and plumb that along here to know which rotation state we're in maybe?

Reply:
I feel bad for trying to merge this. I was pretty sure it was incorrect. Thanks for quickly finding a counter example.

Reply:
Thanks for writing a comment expressing doubt :)  Makes finding bugs really easy

Reply:
Fixed! Note this breaks the serialization format, so you'll have to delete/recreate any projects that have diagonal filters.